### PR TITLE
types: fqdn: allow dashes, too!

### PR DESCRIPTION
**Pull request by williballenthin**

### CODE REVIEW

**Comment by invisig0th** [comment]
Sweet, thanks. I'll merge this as soon as I'm in front of my workstation.

File: synapse/lib/types.py [hunk 1]
```diff
@@ -507,7 +507,7 @@ def __init__(self):
         self.addSubType('inet:passwd','str')
         self.addSubType('inet:filepath','str')
 
-        self.addSubType('inet:fqdn','str', regex='^[a-z0-9.-_]+$', lower=1)
+        self.addSubType('inet:fqdn','str', regex='^[a-z0-9.-_\-]+$', lower=1)
```
Comment (aheinz-fe) [inline synapse/lib/types.py]: I think it may be reading `.-_` as a range? Maybe just escape that dash instead of adding another?

Comment (williballenthin) [inline synapse/lib/types.py]: i think you're right. tbh, i also read it as the range of characters between `.` and `_` and assumed this made sense. reviewing my ascii table, this doesnt make any sense. so, a better fix is as you suggested: to escape the dash.

**Comment by invisig0th** [comment]
fixed in separate commit

**Pull request closed without merging**